packages/sdk/tests/e2e/dataProtectorSharing/consumeProtectedData.test.ts

```diff
@@ -90,24 +70,106 @@ describe.skip('dataProtector.consumeProtectedData()', () => {
+        };
         // --- WHEN
-        const onStatusUpdateMock = jest.fn();
-        const result = await dataProtectorConsumer.sharing.consumeProtectedData(
-          {
+        await expect(
+          dataProtectorConsumer.sharing.consumeProtectedData({
             app: DEFAULT_PROTECTED_DATA_DELIVERY_APP,
             protectedData,
             workerpool: WORKERPOOL_ADDRESS,
-            onStatusUpdate: onStatusUpdateMock,
-          }
+            maxPrice: 0,
+            onStatusUpdate: onStatusUpdate,
+          })
+        ).rejects.toThrow(
+          `This whitelist contract does not have registered this app: ${DEFAULT_PROTECTED_DATA_DELIVERY_APP.toLocaleLowerCase()}.`
         );
-
         // --- THEN
-        expect(onStatusUpdateMock).toHaveBeenCalledWith({
-          title: 'CONSUME_RESULT_COMPLETE',
-          isDone: true,
+      },
+      timeouts.protectData +
+        timeouts.createCollection +
+        timeouts.addToCollection +
+        timeouts.setProtectedDataToSubscription +
+        timeouts.setSubscriptionParams +
+        timeouts.subscribe
+    );
+  });
+
+  describe('When whitelist contract registered delivery app', () => {
+    beforeAll(async () => {
+      // add app 'protected-data-delivery-dapp-dev.apps.iexec.eth' to AppWhitelist SC
+      await dataProtectorCreator.sharing.addAppToAddOnlyAppWhitelist({
+        addOnlyAppWhitelist,
+        app: DEFAULT_PROTECTED_DATA_DELIVERY_APP,
+      });
+    });
+
+    it(
+      'should work when calling consumeProtectedData() with valid inputs',
+      async () => {
+        // --- GIVEN
+        const { address: protectedData } =
+          await dataProtectorCreator.core.protectData({
+            data: { file: 'test' },
+            name: 'consumeProtectedDataTest',
+          });
+        const { collectionId } =
+          await dataProtectorCreator.sharing.createCollection();
+
+        await dataProtectorCreator.sharing.addToCollection({
+          collectionId,
+          addOnlyAppWhitelist,
+          protectedData,
+        });
+        await dataProtectorCreator.sharing.setProtectedDataToSubscription({
+          protectedData,
+        });
+        // TODO : add test with price !== 0
+        const subscriptionParams = { price: 0, duration: 86400 };
+        await dataProtectorCreator.sharing.setSubscriptionParams({
+          collectionId,
+          ...subscriptionParams,
+        });
+
+        await dataProtectorConsumer.sharing.subscribeToCollection({
+          collectionId,
+          ...subscriptionParams,
         });
-        console.log(result);
+
+        await expect(
+          dataProtectorConsumer.sharing.consumeProtectedData({
+            protectedData,
+            app: DEFAULT_PROTECTED_DATA_DELIVERY_APP,
+          })
+        ).rejects.toThrow(
+          new Error('Sharing smart contract: Failed to consume protected data')
+        );
+        // TODO: Uncomment this part once a workerpool is deployed locally in TEE.
```

Comment:
Not sure we'll be able to run that one day.
But at least we understand, ok to keep this commented code.